packages/block-library/src/social-link/edit.native.js

```diff
@@ -67,9 +69,14 @@ const SocialLinkEdit = ( {
 
 	// When new social icon is added link sheet is opened automatically
 	useEffect( () => {
+		updateSocialIconName( true );
 		if ( isSelected && ! url ) {
 			setIsLinkSheetVisible( true );
 		}
+
+		return () => {
+			updateSocialIconName( false );
+		};
```

Comment:
* What happens when the post is saved while the social icon is being edited?
* What happens if I try to _Undo_ right after adding the social icon? (risk of _undo trap_, #8119)

Reply:
> * What happens when the post is saved while the social icon is being edited?
Works properly

> * What happens if I try to Undo right after adding the social icon? (risk of undo trap, #8119)

Good catch, there is a _undo trap_ :( Do you have any suggestion how to avoid that?

Reply:
Basically I need to have full social icon name e.g.: `core/social-link-wordpress`, so I have one more idea to extend selector `getBlockName` to return name with service for native platforms:

```
export function getBlockName( state, clientId ) {
	const block = state.blocks.byClientId[ clientId ];

	if ( Platform.OS !== 'web' && block?.name === 'core/social-link' ) {
		const attributes = state.blocks.attributes[ clientId ];
		const { service } = attributes;

		return service ? `core/social-link-${ service }` : block.name;
	}
	return block ? block.name : null;
}
```

Reply:
> Good catch, there is an undo trap :( Do you have any suggestion how to avoid that?

In general, it's a strong anti-pattern to trigger an attribute change that isn't a direct consequence of a user interaction. For example: the user types a letter, it updates the `content` attribute; the user checks "Enable drop cap", that toggles `dropCap`; etc. These are all clean scenarios that don't affect the ability to undo.

The _undo trap_ issues always arise from automated attribute setting. For example: while a Gallery block is loading with partial attributes (e.g. `{ images: [ { id: 42 } ] }`, it calls `getMedia` to get more data, and automatically updates the attributes once the request resolves (`{ images: [ { id: 42, src: '...', alt: '...' } ] }`). The present case with Social Icons is identical, but without even a request.

In all these cases, there's a call to `setAttributes` that arises not from an _event_ (e.g. user input), but — directly or indirectly — from a _state_:
* Directly: state is `blockName === 'core/social-link'` **->** call `setState`
* Indirectly: state is missing image data **->** make REST API request -> call `setState`

> so I have one more idea to extend selector getBlockName to return name with service for native platforms

Yeah, that might work, as long as there is never a change to attributes (e.g. to touch `service`). And, naturally, with the understanding that this is a patch until you can confidently let the mobile editor handle variations of `core/social-link` naturally. :)